File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        curr = None
        i = 0

        while i < len(nums):
            if nums[i] == curr:
                nums.pop(i)
            else:
                curr = nums[i]
                i += 1
            
        return len(nums)

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int len = nums.size();
        vector<int> maxSubsequences(len, 1);
        int retVal = 1;

        for (int i = len - 2; i >= 0; i--) {
            int curr = nums[i];
            int l = 1;

            for (int j = i + 1; j < len; j++) {
                if (nums[j] > curr) {
                    l = max(l, 1 + maxSubsequences[j]);
                }
            }

            maxSubsequences[i] = l;
            retVal = max(retVal, l);
        }

        return retVal;
    }
};
